### PR TITLE
Add a new example `max_trial_callback` to `optuna/examples` 

### PR DESCRIPTION
**Pull request by jeromepatel**
## Motivation
With reference to issue #1883, I have added a new example regarding the discussion on that issue. 

## Description of the changes
 Currently, I have implemented a very simple <code>objective</code> function which uses <code>max_trial_callback</code> for checking maximum number of trial, including completed as well as running.

**Note**:
This implementation is still very minimal, but I think there is some room for improvements, such as using <code>LinearRegression</code> from sklearn with Iris dataset (or any other dataset) and printing more information to the user. I am open to discussion and would like to know everyone's views on this example. 
I also have fixed a minor typo in another example file <code>simple_prunning.py</code>.

### CODE REVIEW

**Comment by jeromepatel** [comment]
If any contributor could guide me and discuss the approach, it would be helpful. Thank you for your valuable time.

File: examples/max_trials_callback.py
Comment (nzw0301) [inline examples/max_trials_callback.py:5]:
```suggestion
In this example, we optimize a simple quadratic function. We use multiple

```

**Comment by jeromepatel** [comment]
@nzw0301 Thank you for reviewing, I have fixed the typo. Let me know your views on whether I should implement a more complex model (for eg. SVC on Iris) or current minimal implementation is fine in the context of this example.

**Comment by nzw0301** [comment]
I think the current code is enough to understand the basic usage to stop the distributed optimisation by the callback. I'm worried that the `objective` is too simple to run another python process for distributed optimisation. So I suggest using a more time-consuming example or the` time.sleep` method to run another process.

File: examples/max_trials_callback.py
Comment (nzw0301) [inline examples/max_trials_callback.py:37]: `create_study` can load exising study with `load_if_exists=True`.

```suggestion
    study = optuna.create_study(
        study_name="test",
        storage="sqlite:///database.sqlite",
        load_if_exists=True
    )

```

File: examples/max_trials_callback.py
Comment (nzw0301) [inline examples/max_trials_callback.py:39]:
```suggestion
```

**Comment by nzw0301** [comment]
How about  printing the `best_value` and `study.best_params` and `len(trials[trials.state == "COMPLETE"])`, and removing `assert` line?

File: examples/max_trials_callback.py
Comment (jeromepatel) [reply examples/max_trials_callback.py:39]: Thank you for reviewing, I have made suggested changes.

**Comment by jeromepatel** [comment]
@nzw0301 I agree with you, printing the `best_value`, `study.best_params`, and `len(trials[trials.state == "COMPLETE"])`  can give user a better understanding of how the example works out. I'll incorporate changes in next commit. Thank you for reviewing.

File: examples/max_trials_callback.py
Comment (nzw0301) [inline examples/max_trials_callback.py:16]: The variable name sounds unclear. How about `max_completed_trials`?

**Comment by nzw0301** [comment]
@jeromepatel  Awesome! I'm happy with the example except for `objective`. Let's hear the opinion about it from the core-members.

**Comment by HideakiImamura** [comment]
@Crissman Could you review this PR if you have time?

File: examples/max_trials_callback.py
Comment (Crissman) [inline examples/max_trials_callback.py:2]:
```suggestion
Optuna example to demonstrate setting the maximum number of trials in

```

File: examples/max_trials_callback.py
Comment (Crissman) [inline examples/max_trials_callback.py:7]:
```suggestion
which allows the user to set a maximum number of trials

```

File: examples/max_trials_callback.py
Comment (Crissman) [inline examples/max_trials_callback.py:6]:
```suggestion
script runs (workers) to demonstrate the use of max_trial_callbacks,

```

**Review by Crissman — CHANGES_REQUESTED** [review]
Thanks for the example!

**Comment by jeromepatel** [comment]
@Crissman Thank you for your comments. I have made the suggested changes. PTAL.

**Review by Crissman — APPROVED** [review]
LGTM!

**Comment by HideakiImamura** [comment]
@nzw0301 Does it look good to merge this PR already? If so, I'd appreciate it if you could approve it.

**Review by nzw0301 — APPROVED** [review]

**Comment by nzw0301** [comment]
@HideakiImamura It does. I have approved this PR.

@jeromepatel Thank you for this good example!

**Comment by Crissman** [comment]
@nzw0301 Please look at the `Files Changed` tab, open `Review Changes` and select `Approve` to record your approval. 🙏

**Review by nzw0301 — APPROVED** [review]

**Comment by nzw0301** [comment]
I'm afraid but I have already done... https://github.com/optuna/optuna/pull/2449#pullrequestreview-612904784
I don't have the right to write access, so that is reason why my approve does not look like another dev member's one.

**Pull request merged**
